### PR TITLE
Allowing field mappings

### PR DESCRIPTION
**Pull request by fidesachates**
@sksamuel 

Couldn't find this feature: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-field-mapping.html

Originally I had opened a pr that merges this functionality into the existing `GetMappingRequest` and modified the respective `MappingHandler`. I realized that at the end of the day, I judged the two requests and responses to be just different enough to warrant a separate `Request` and thus a separate `Handler`. 

This PR also removes `local` method from `GetMappingRequest` as I couldn't find any usage of it

### CODE REVIEW

File: elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala [hunk 1]
```diff
@@ -74,5 +84,55 @@ class MappingHttpTest extends WordSpec with DockerTests with Matchers with Befor
...
+        getMapping("indexnopropsempty")
+      }.await.result
+
+      val properties = mappings.find(_.index == "indexnopropsempty").get.mappings
+
+      properties shouldBe Map.empty
+    }
+  }
+
+  "field mapping get" should {
```
Comment (fidesachates) [inline elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala]: Test cases for when we ask for a specific field in a mapping get request. These are a mirror of the original two tests in addition to the third one I added for when we get all mappings.

File: elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala [hunk 2]
```diff
@@ -74,5 +84,55 @@ class MappingHttpTest extends WordSpec with DockerTests with Matchers with Befor
 
       properties shouldBe Map.empty
     }
+
+    "handle properly completely empty mapping" in {
```
Comment (fidesachates) [inline elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala]: New edge case test when the mapping is completely empty

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala [hunk 3]
```diff
@@ -28,12 +32,49 @@ trait MappingHandlers {
...
+  implicit object GetFieldMappingRequest extends Handler[GetFieldMappingRequest, Seq[IndexFieldMapping]] {
+    override def responseHandler: ResponseHandler[Seq[IndexFieldMapping]] = new ResponseHandler[Seq[IndexFieldMapping]] {
+
+      override def handle(response: HttpResponse) : Either[ElasticError, Seq[IndexFieldMapping]] = response.statusCode match {
+        case 201 | 200       =>
+          val raw = ResponseHandler.fromResponse[Map[String, Map[String, Map[String, Map[String, Any]]]]](response)
+          Right(raw.map {
+            case (index, types) =>
+              val mappings = types("mappings").map {
+                case (_, mapping) => FieldMapping(mapping("full_name").toString, mapping("mapping").asInstanceOf[Map[String, Any]])
```
Comment (fidesachates) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala]: This method is largely a copy pasta, but this part is different as it's deserializing an object that looks like this:

```
{
    "index": {
        "mappings": {
            "a": {
                "full_name": "a",
                "mapping": {
                    "a": {
                        "type": "text",
                        "store": true,
                        "analyzer": "whitespace"
                    }
                }
            }
        }
    }
}
```

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/mappings/GetMappingRequest.scala [hunk 4]
```diff
@@ -1,8 +1,5 @@
 package com.sksamuel.elastic4s.requests.mappings
 
 import com.sksamuel.elastic4s.Indexes
-import com.sksamuel.exts.OptionImplicits._
 
-case class GetMappingRequest(indexes: Indexes, local: Option[Boolean] = None) {
-  def local(local: Boolean): GetMappingRequest = copy(local = local.some)
```
Comment (fidesachates) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/mappings/GetMappingRequest.scala]: As I said, removing because I couldn't find a usage of it.

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala [hunk 5]
```diff
@@ -28,12 +32,49 @@ trait MappingHandlers {
...
+    }
+
+    override def build(request: GetFieldMappingRequest): ElasticRequest = {
+      val baseEndpoint = request.indexes match {
+        case Indexes(Nil)       => "/_mapping"
+        case Indexes(indexes)   => s"/${indexes.mkString(",")}/_mapping"
+      }
+
+      val endpoint = request.fields match {
+        case Nil => throw new IllegalArgumentException("Empty fields in GetFieldMappingRequest. Use GetMappingRequest instead.")
```
Comment (fidesachates) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala]: Perhaps this isn't the right thing to do? A field mapping request requires at least one field;.

**Comment by sksamuel** [comment]
Sorry for the late merge, but great work.
